api/src/test/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAOTest.java

```diff
@@ -0,0 +1,168 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.AdministrationDAO;
+import org.hibernate.SessionFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Location;
+import org.openmrs.Role;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.openmrs.test.Verifies;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.BindException;
+import org.springframework.validation.Errors;
+
+public class HibernateAdministrationDAOTest extends BaseContextSensitiveTest {
+
+	@Autowired
+	private HibernateAdministrationDAO dao;
+
+	private SessionFactory sessionFactory;
+
+	@Before
+	public void getSessionFactory() throws Exception {
+		sessionFactory = (SessionFactory) applicationContext.getBean("sessionFactory");
+	}
+
+
+
+	/**
+	 * @see HibernateAdministrationDAO#validate(Object,Errors)
+	 * @verifies Fail validation for location class if field lengths are not correct
+	 */
+	@Test
+	public void validate_shouldFailValidationForLocationClassIfFieldLengthsAreNotCorrect() throws Exception {
+		Location location = new Location();
+		String longString = "too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text";
+		location.setName(longString);
+		location.setDescription(longString);
+		location.setAddress1(longString);
+		location.setAddress2(longString);
+		location.setAddress3(longString);
+		location.setAddress4(longString);
+		location.setAddress5(longString);
+		location.setAddress6(longString);
+		location.setAddress7(longString);
+		location.setAddress8(longString);
+		location.setAddress9(longString);
+		location.setAddress10(longString);
+		location.setAddress11(longString);
+		location.setAddress12(longString);
+		location.setAddress13(longString);
+		location.setAddress14(longString);
+		location.setAddress15(longString);
+		location.setCityVillage(longString);
+		location.setStateProvince(longString);
+		location.setCountry(longString);
+		location.setPostalCode(longString);
+		location.setLatitude(longString);
+		location.setLongitude(longString);
+		location.setCountyDistrict(longString);
+		location.setRetireReason(longString);
+		
+		Errors errors = new BindException(location, "location");
+		dao.validate(location, errors);
+		
+		Assert.assertTrue(errors.hasFieldErrors("name"));
```

Comment:
I think it would be nice to check the actual error message code because the field might be invalid but because of something else, you can loop over all the fields since the error message is expected to be the same